src/com/google/enterprise/adaptor/fs/FsAdaptor.java

```diff
@@ -1086,81 +1093,61 @@ private void getFileAcls(Path doc, Response resp) throws IOException {
     }
   }
 
-  /* Makes HTML document with web links to namespace's DFS links. */
-  private void getDfsNamespaceContent(Path doc, DocId docid, Response resp)
-      throws IOException {
-    // TODO (bmj): Handle large Namespaces by paying attention to maxHtmlLinks
-    // and sending excess to DocIdPusher.
-    resp.setNoIndex(!indexFolders);
-    try (HtmlResponseWriter writer = createHtmlResponseWriter(resp)) {
-      writer.start(docid, getFileName(doc));
-      try (DirectoryStream<Path> links = delegate.newDfsLinkStream(doc)) {
-        for (Path link : links) {
-          DocId docId;
-          try {
-            docId = delegate.newDocId(link);
-          } catch (IllegalArgumentException e) {
-            log.log(Level.WARNING, "Skipping DFS link {0} because {1}.",
-                    new Object[] { link, e.getMessage() });
-            continue;
-          }
-          writer.addLink(docId, getFileName(link));
-        }
-      }
-      writer.finish();
-    }
-  }
-
-  /* Makes HTML document with links to this directory's files and folders. */
-  private void getDirectoryContent(Path doc, DocId docid,
-      FileTime lastAccessTime, Response resp) throws IOException {
+  /* Makes HTML document with links to this DirectoryStream's contents. */
+  private void getDirectoryStreamContent(Path source, DocId docid,
+       FileTime lastAccessTime, Response resp, DirectoryStreamFactory factory)
+       throws IOException {
     resp.setNoIndex(!indexFolders);
 
     // Large directories can have tens or hundreds of thousands of files.
     // The GSA truncates large HTML documents at 2.5MB, so return the first
     // maxHtmlLinks worth as HTML content and send the rest to the DocIdPusher.
-    try (DirectoryStream<Path> files = delegate.newDirectoryStream(doc);
+    try (DirectoryStream<Path> paths = factory.newDirectoryStream(source);
          HtmlResponseWriter htmlWriter = createHtmlResponseWriter(resp)) {
-      htmlWriter.start(docid, getFileName(doc));
+      htmlWriter.start(docid, getFileName(source));
       int htmlLinks = 0;
-      for (Path file : files) {
-        DocId docId = delegate.newDocId(file);
+      for (Path path : paths) {
+        DocId docId = delegate.newDocId(path);
         if (htmlLinks++ < maxHtmlLinks) {
           // Add an HTML link.
-          htmlWriter.addLink(docId, getFileName(file));
+          htmlWriter.addLink(docId, getFileName(path));
         } else {
           String message = MessageFormat.format(
-              "Directory listing for {0} exceeds maxHtmlSize of {1,number,#}. "
-              + "Switching to asynchronous feed of directory content.",
-              doc, maxHtmlLinks);
+              "Content listing for {0} exceeds maxHtmlSize of {1,number,#}. "
+              + "Switching to asynchronous feed of content.",
```

Comment:
Maybe s/content/children/ or "child docids". Here and elsewhere, I'm a little nervous about the conflation of directory listings with file contents in the log messages. I think it's less confusing to call a DFS link a "directory" than to call a directory listing "content".